index.js

```diff
@@ -37,6 +37,23 @@ const pDebounce = (fn, wait, options = {}) => {
 	};
 };
 
+pDebounce.promise = function_ => {
+	let currentPromise;
+
+	return async (...arguments_) => {
+		if (currentPromise) {
+			return currentPromise;
+		}
+
+		try {
+			currentPromise = function_(...arguments_);
+			return await currentPromise;
+		} finally {
+			currentPromise = undefined;
+		}
```

Comment:
Suggested non-async code
```suggestion
	  return currentPromise.then(result => {
	 	  currentPromise = undefined;
		  return result;
	  }, error => {
	 	  currentPromise = undefined;
		  throw error;
	  });
```